src/match_identifiers.cpp

```diff
@@ -23,6 +23,10 @@ void match_identifiers(TypedAST::Declaration* ast, Frontend::CompileTimeEnvironm
 	env.declare(ast->identifier_text(), ast);
 	if (ast->m_value) {
 		match_identifiers(ast->m_value.get(), env);
+		// TODO: should this be a Var instead of just an id copy?
+		ast->m_value_type = ast->m_value->m_value_type;
```

Comment:
Sí. Si hacés una copia, y después el type id del rhs lo pisás al hacer el union find, después el valor del decl queda desactualizado. Sería mejor el Var así dice "tengo el mismo tipo que el rhs"

Reply:
I don't think this is how that works